src/hooks/useTokensIdentityRegistries.ts

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { TokenNameItem } from "@/types/types";
+import { WatchContractEventReturnType } from "@/services/contracts/watchContractEvent";
+import {
+  readTokenIdentityRegistry,
+  watchIdentityRegistryIdentityRegisteredEvent,
+} from "@/services/contracts/wagmiGenActions";
+
+export function useTokensIdentityRegistries(tokens?: TokenNameItem[]) {
```

Comment:
Why tokens are optional?

Reply:
> Why tokens are optional?

For the same proposal as tokens could be not selected at all:
https://github.com/web3-nomad/accelerator-defi-EIP-ui/blob/153ecfcb7d819965e02d0b4d40a8b6ecfd81b6b6/src/hooks/useTokenIdentityRegistry.ts#L16